posture/serialnumber_notmacos.go

```diff
@@ -71,7 +72,31 @@ func init() {
 	numOfTables = len(validTables)
 }
 
-func GetSerialNumbers(logf logger.Logf) ([]string, error) {
+func GetSerialNumbers(logf logger.Logf) (serials []string, err error) {
+	defer func() {
```

Comment:
looks sensible to me, is there any particular reason to do it in a defer instead of the end of the function?

Reply:
The idea was that `GetSerialNumbers` should always succeed if a serial number is configured via syspolicy. Using defer allows us to always check the policy whether reading from SMBIOS fails or succeeds.